libcperciva/util/sock.c

```diff
@@ -406,13 +406,35 @@ sock_connect(struct sock_addr * const * sas)
  */
 int
 sock_connect_nb(const struct sock_addr * sa)
+{
+
+	/* Let sock_connect_bind_nb handle this. */
+	return (sock_connect_bind_nb(sa, NULL));
+}
+
+/**
+ * sock_connect_bind_nb(sa, sa_b):
+ * Create a socket, mark it as non-blocking, and attempt to connect to the
+ * address ${sa}.  If ${sa_b} is not NULL, then bind the socket to ${sa_b}.
```

Comment:
I think that this is what @cperciva asked for, but having a function called `sock_connect_bind_nb()` and then the comment `If ${sa_b} is not NULL, then bind the socket to ${sa_b}` strikes me as a bit weird.

If we're going with 3 different `network_connect` public functions (which all call `network_connect_internal`), then should we do a similar thing in `sock.c`?  i.e. add a `sock_connect_nb_internal`, which is called by `sock_connect_nb` and `sock_connect_bind_nb`?

Reply:
Let's keep it as-is for now.  If we end up creating a third "flavour" of `sock_connect_nb` then we can go down the `_internal` route at that point.